src/jit/emitarm.cpp

```diff
@@ -7787,6 +7787,17 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
             assert(!"Invalid ins for overflow check");
         }
     }
+
+    if (dst->gtSetFlags())
+    {
+        if ((ins != INS_cmp) && (ins != INS_cmn))
```

Comment:
That was a note to self to remove the unnecessary `if`. I don't think there's any doubt that `cmp` is not ternary :)